### DISCUSSION TITLE
Import templates from another YAML file

### DISCUSSION THREAD

**Discussion by franciscoescher** [argoproj/argo-workflows#9136] [Q&A] [answered]
Hello,

Recently I've been using argo to implement a data pipeline that is quite complex, and I've been struggling to handle modularity in a direct and easy way. The main reason for that is that I can not have my different templates stored in different files, so I end up with a workflow yaml file that is simply too big.

I know that I can use workflow templates with the templateRef for that, but those first need to be deployed to the cluster. In a scenario that there is a lot of developers working in parallel, managing those deployed workflows becomes a little annoying, and also gives me the headache of dealing with versioning of the different workflow templates and more deployment processes overhead.

Wouldn't it be super good to allow templates to be structured as a proper code language, and be imported or referenced locally, when the submit command is submitted?

Note: Originally I've posted this into the https://github.com/argoproj
/
[argo-cd](https://github.com/argoproj/argo-cd) discussions and was told to post it here, and was also given the idea to use helm and the python library

**Comment by terrytangyuan** [comment, marked as answer]
You need configuration management tools such as kutomize, ytt, or use languages like Python to organize and modularize your templates.
